src/components/CompactDashboard/CompactDashboard.tsx

```diff
@@ -0,0 +1,197 @@
+
+import * as React from 'react';
+import { ICompactDashboardProps, IFarm } from './CompactDashboard.Props';
+import { CompactServer } from '../CompactServer/CompactServer';
+import { Group } from '../Group/Group';
+import { GroupHeader } from '../GroupHeader/GroupHeader';
+const List = require('react-virtualized').List;
+const AutoSizer = require('react-virtualized').AutoSizer;
+const Collection = require('react-virtualized').Collection;
+import * as classNames from 'classnames';
+import { autobind } from '../../utilities/autobind';
+import './CompactDashboard.scss';
+
+const GUTTER_SIZE = 3;
+const CELL_WIDTH = 330;
+
+function sortFarmServers(ob1: { status: number, serverName: string }, ob2: { status: number, serverName: string }) {
+    if (ob1.status > ob2.status) {
+        return 1;
+    } else if (ob1.status < ob2.status) {
+        return -1;
+    }
+
+    if (ob1.serverName < ob2.serverName) {
+        return -1;
+    } else if (ob1.serverName > ob2.serverName) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+function sortFarms(ob1: { farmName: string }, ob2: { farmName: string }) {
+    if (ob1.farmName < ob2.farmName) {
+        return -1;
+    }
+
+    if (ob1.farmName > ob2.farmName) {
+        return 1;
+    }
+    return 0;
+}
+
+function checkFilter(filter: string, serverName: string) : boolean {
+    return serverName.toLowerCase().trim().indexOf(filter.toLowerCase().trim()) !== -1;
+}
+
+export class CompactDashboard extends React.Component<ICompactDashboardProps, any> {
+
+    collection: any;
+    list: any;
+
+    constructor(props?: ICompactDashboardProps) {
+        super(props);
+        this.state = { columnYMap: [], collection: undefined, list: undefined };
+    }
+
+    @autobind
+    private componentDidUpdate(prevProps: ICompactDashboardProps, prevState) {
+        if (this.props.isVertical === false && prevProps.isVertical === true) {
+            this.setState({ columnYMap: [] });
+        }
+        if (this.props.filter !== prevProps.filter) {
+            if (this.collection) {
+                this.setState({ columnYMap: [] }, () => {
+                    this.collection.recomputeCellSizesAndPositions();
+                });
+            }
+            if (this.list) {
+                this.list.recomputeRowHeights();
+            }            
+        }       
+    }
+
+
+    public render() { 
+        let {title, farms} = this.props;
+        let classname = classNames({ [this.props.className]: this.props.className !== undefined });
+        return (
+            <div className={classname}>
+                {
+                    this.props.isVertical &&
+                    <div className="compact-dashboard-container vertical">

```

Comment:
I don't think it should. It woudn't serve any purpose other than 'cleaning-up' this look of the class. It won't be reusable due to the fact that each project will need to implement it's custom logic in handling react-virtualized components (if there is the need for some other components to be placed inside)